contracts/LockDealNFT/LockDealNFT.sol

```diff
@@ -94,15 +94,15 @@ contract LockDealNFT is LockDealNFTModifiers, ILockDealNFTEvents {
         address newOwner
     ) external onlyOwnerOrAdmin(poolId) {
         address provider = poolIdToProvider[poolId];
-        uint256 dataPoolId;        
-        // refund provider case
-        if(poolIdToVaultId[poolId] == 0 && !approvedProviders[msg.sender]) {
-            newOwner = provider;
-            dataPoolId = poolId - 2;
-        }
-        else {
-            dataPoolId = poolId;
-        }
+        uint256 dataPoolId = poolId;        
+        // // refund provider case
+        // if(poolIdToVaultId[poolId] == 0 && !approvedProviders[msg.sender]) {
```

Comment:
In this case, the `LockDealBundleProvider` contract does not match the conditions.
```solidity
if(!approvedProviders[msg.sender])
{
       // LockDealBundleProvider can't be here
}
```

But in any case, the next version of the pefund provider should change the if logic.

Reply:
The`split` function on LockDealNFT is called by the user, so `LockDealBundleProvider` contract match the condition.
Workflow: User -> Split(LockDealNFT) -> Split(LockDealBundleProvider)
Here is the test you can take a look at.
https://github.com/The-Poolz/LockDealNFT/blob/6b6172f432e9d45c44e3e852409caa6684acb24b/test/LockDealProvider.ts#L83
So if you can add any specific condition for the refund provider, that would be helpful.

Reply:
compromise solution: https://github.com/The-Poolz/LockDealNFT/pull/138